source_pubsub/client.go

```diff
@@ -255,39 +261,67 @@ func (c *PubSubClient) Wait(ctx context.Context) error {
 	}
 }
 
+func (c *PubSubClient) ResetRetryCount() {
+	c.retryCount = c.maxRetries
+}
+
 func (c *PubSubClient) ReplayID() []byte {
 	return c.currReplayID
 }
 
+func (c *PubSubClient) retryAuth(ctx context.Context, retry bool) (error, bool) {
```

Comment:
As an example, if you use `retry.Do` here and use https://github.com/avast/retry-go/blob/01d7e0007ca266461dd395bc76ee7d14b2a73b63/options.go#L182-L214 it can simply the code

E.g.

```
retry.Do(
  func() error {
     if err := login(ctx)... {
        return err
     }
     if err := c.canSubscribe ... {
       return err
     }
  },
  rt.Attempts(10),
  ...
)
```

You can set the number of tries to attempt. If it returns error after 10, exit if not next time it needs to retry it will do 10 more times again, you can configure back off and delay in between.

Reply:
Added!!